app/controllers/concerns/books_controller.rb

```diff
@@ -0,0 +1,17 @@
+class BooksController < ApplicationController
+
+def show
+  @book = Book.find(params[:id])
+  @reviews = @book.reviews
+  @average = @book.average
+  @highest_rating = @book.highest_rating
+  @lowest_rating = @book.lowest_rating
+  @high_review= @book.highest_rating_review
+  @low_review= @book.lowest_rating_review
+  # @high_user = User.find(@high_review.user_id)
+  # @low_user = User.find(@low_review.user_id)
+
+# end
+end
```

Comment:
Your ultimate goal is to only pass one instance variable to the view.  It can definitely get tricky to do this, and you'll learn about some more patterns to accomplish this goal in Mod3. Here though, you could pass @book to the view and from the view call @book.average etc. 

You might have not done this in trying to get logic out of the view, but it is considered okay to call a method from an object in a view. In this sense @book.average is as much logic as @book.title. I'd expect this controller action to only have @book = Book.find(params[:id]) in it.

Reply:
You may have been using this strategy since you were getting Nil errors, but I wouldn't expect to see a User.find here. I'd want to scope from the book to the review to the user for that review. This logic for this should be in the model.